### PR TITLE
ListStyle -> ListLength and add _with_length

### PR DESCRIPTION
**Pull request by sffc**
Part of #2136

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]

File: components/list/src/list_formatter.rs [hunk 1]
```diff
@@ -47,23 +47,23 @@ macro_rules! constructor {
 
 impl ListFormatter {
     constructor!(
-        try_new_and_unstable,
-        try_new_and_with_any_provider,
-        try_new_and_with_buffer_provider,
+        try_new_and_with_length_unstable,
```
Comment (robertbastian) [inline components/list/src/list_formatter.rs:50]: Please let's not list every parameter in the function name, this is not Objective-C. I don't know why you want to change this for FFI, but that can be done independently of the Rust names.

**Review by robertbastian — DISMISSED** [review]

File: components/list/src/list_formatter.rs (hunk 1 above)
Comment (sffc) [reply components/list/src/list_formatter.rs:50]: I was doing it for every type that doesn't take an options bag. If we want to add `ListFormatterOptions`, then that function would get the nice `try_new` name. But in the absence of such a type, we use a more specific function name. I did it already for DateTimeFormatter and now for ListFormatter. @zbraniecki

Comment (robertbastian) [reply components/list/src/list_formatter.rs:50]: "I did it already for DateTimeFormatter" isn't very convincing. The arguments are already part of the function signature, they don't need to be repeated in the name. Any place that shows the name (docs.rs, autocomplete) also shows the arguments.

Here especially, I don't see any other possible options, so overload-busting is not an argument.

Comment (Manishearth) [reply components/list/src/list_formatter.rs:50]: Yeah, I think we already kinda settled on the convention of doing one of (based on the situation):

 - Take an options bag, and have a short name
 - Take specific things, and be 100% sure they will never change
 - Take specific things and name the function after them. If more things are added we can go to an options bag and deprecate the old API

Unless we're 100% sure of there being no future options (I don't think we are) then we should probably merge as is.

Comment (Manishearth) [reply components/list/src/list_formatter.rs:50]: I feel like this was already discussed and decided on as a convention a while back, I'm not a fan of switching gears last moment.

Comment (Manishearth) [reply components/list/src/list_formatter.rs:50]: I'd recommend merging this, if we are convinced we'll never have an options bag here we can deprecate the function and add a new one post 1.0, this is too close to 1.0 to debate now.

Comment (zbraniecki) [reply components/list/src/list_formatter.rs:50]: My position has always been to use Options Bag for all component main constructors in Rust.

Comment (sffc) [reply components/list/src/list_formatter.rs:50]: Giving the function a long name is easy to back out (we can doc-hidden reexport it). The other direction is not as easily reversible.

**Pull request merged**